src/AppInstallerCLICore/ExecutionContext.h

```diff
@@ -101,6 +101,10 @@ namespace AppInstaller::CLI::Execution
             WI_ClearAllFlags(m_flags, flags);
         }
 
+        void SetExecutionStage(Workflow::ExecutionStage stage, bool allowBackward);
+        virtual void SetExecutionStage(Workflow::ExecutionStage stage);

```

Comment:
No need to separate these functions, but if you are going to do so, the virtual function should be protected.